audmath/core/api.py

```diff
@@ -78,6 +87,141 @@ def db(
     return x
 
 
+def window(
+    samples: int,
+    shape: str = 'tukey',
+    half: str = None,
+) -> np.ndarray:
+    r"""Return a window.
+
+    The window will start from
+    and/or end at 0.
+    If at least 3 samples are requested
+    and the number of samples is odd,
+    the windows maximum value will always be 1.
+
+    The shape of the window
+    is selected via ``shape``
+    The following figure shows all available shapes.
+    For the Kaiser window
+    we use :math:`\beta = 14`
+    and set its first sample to 0.
+
+    .. plot::
+
+        import audmath
+        import matplotlib.pyplot as plt
+        import matplotlib.ticker as mtick
+        import numpy as np
+        import seaborn as sns
+
+        for shape in audmath.core.api.WINDOW_SHAPES:
+            win = audmath.window(101, shape=shape)
+            plt.plot(win, label=shape)
+        plt.ylabel('Magnitude')
+        plt.xlabel('Window Length')
+        plt.grid(alpha=0.4)
+        ax = plt.gca()
+        ax.xaxis.set_major_formatter(mtick.PercentFormatter())
+        ax.tick_params(axis=u'both', which=u'both',length=0)
+        plt.xlim([-1.2, 100.3])
+        plt.ylim([-0.02, 1])
+        sns.despine(left=True, bottom=True)
+        # Put a legend to the top right of the current axis
+        plt.legend()
+        ax.legend(loc='upper left', bbox_to_anchor=(1, 1))
+        # Adjsut image size to contain outside legend
+        fig = plt.gcf()
+        fig.set_size_inches(6.4, 3.84)
+        plt.tight_layout()
+
+    Args:
+        samples: length of window
+        shape: shape of window
+        half: if ``None`` return whole window,
+            if ``left`` or ``right``
+            return left or right window
+            inlcuding center
```

Comment:
I guess I need to reformulate this. At the moment if you specify `left` or `right` the half-window will always include 1 as maximum, whereas if you specify `half=None` and request a window with an odd number of samples you might get a window without 1 as maximum. This means in the case of an odd number of samples we also have `np.concatenate([window(samples / 2, half=left) + window(samples / 2, half=right]) != window(samples)`

Reply:
I decided for this as it is very handy to always get a maximum of 1 for a half window. But if you disagree and it is more important that the two half-windows would combine to the whole window, we can also change this.

Reply:
With the use case we have in mind (applying a half window as fade in / out) it makes sense to include 1 for half windows. So I think we can stick with your solution, but should extend the docstring to be clear about it.

Reply:
I updated the text to

![image](https://user-images.githubusercontent.com/173624/205045231-412168b3-81ec-4eb2-a2c9-23e4623060d1.png)